File: src/main/java/com/store/config/OpenAiProperties.java
package com.store.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "app.ai.openai")
@Data
public class OpenAiProperties {

    private boolean enabled;

    private String apiKey;

    private String model;

    private String baseUrl;

    private Long timeoutSeconds;

}

File: src/main/java/com/store/config/OpenAiProperties.kt
package com.store.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app.ai.openai")
data class OpenAiProperties(
    val isEnabled: Boolean = false,
    val apiKey: String = "",
    val model: String = "",
    val baseUrl: String = "",
    val timeoutSeconds: Long = 10,
)
